### PR TITLE
chore: enable automatic login redirect from app (WEBAPP-5164)

### PR DESCRIPTION
**Pull request by AndyLnd**

### CODE REVIEW

File: app/script/main/app.js [hunk 1]
```diff
@@ -756,14 +756,15 @@ z.main.App = class App {
...
-        let url = `${baseUrl}${location.search}`;
+        let url = `${baseUrl}${window.location.search}`;
         const isImmediateSignOutReason = App.CONFIG.SIGN_OUT_REASONS.IMMEDIATE.includes(signOutReason);
         if (isImmediateSignOutReason) {
           url = z.util.URLUtil.appendParameter(url, `${z.auth.URLParameter.REASON}=${signOutReason}`);
         }
 
         const redirectToLogin = signOutReason !== z.auth.SIGN_OUT_REASON.NOT_SIGNED_IN;
-        if (redirectToLogin) {
+        const forcedAppLogin = z.util.Environment.desktop && z.window.location.hash === '#login';
```
Comment (atomrc) [inline app/script/main/app.js]: What is that `z.window` I don't find it in the codebase?

Comment (atomrc) [reply app/script/main/app.js]: Isn't it supposed to be a `||` here?
If the app is the desktop or that the current location contains `login`?

Comment (AndyLnd) [reply app/script/main/app.js]: Just a typo, already fixed

Comment (AndyLnd) [reply app/script/main/app.js]: I want to keep the '#login' only if it is in the context of the app.

File: app/script/main/app.js [hunk 2]
```diff
@@ -756,14 +756,15 @@ z.main.App = class App {
...
-        let url = `${baseUrl}${location.search}`;
+        let url = `${baseUrl}${window.location.search}`;
         const isImmediateSignOutReason = App.CONFIG.SIGN_OUT_REASONS.IMMEDIATE.includes(signOutReason);
         if (isImmediateSignOutReason) {
           url = z.util.URLUtil.appendParameter(url, `${z.auth.URLParameter.REASON}=${signOutReason}`);
         }
 
         const redirectToLogin = signOutReason !== z.auth.SIGN_OUT_REASON.NOT_SIGNED_IN;
-        if (redirectToLogin) {
+        const forcedAppLogin = z.util.Environment.desktop && window.location.hash === '#login';
```
Comment (atomrc) [inline app/script/main/app.js:766]: Isn't it supposed to be a || here?
If the app is the desktop or that the current location contains login?

Comment (AndyLnd) [reply app/script/main/app.js:766]: Nope, see above.

**Comment by AndyLnd** [comment]
This is related to https://github.com/wireapp/wire-desktop/pull/1607

File: app/script/main/app.js (hunk 2 above)
Comment (gregor) [reply app/script/main/app.js:766]: Do we really need this dependency between wrapper and webapp? Would be nice to not have to think about this in the future with different versions of stuff here and there.

**Pull request closed without merging**